### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD005`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#726]
### Rule details

Add a rule equivalent to markdownlint's MD005.  The rule should report list item markers that are parsed at the same nesting level but are not consistently aligned. It should also support both left- and right-aligned ordered list markers, matching MD005.  Provide an autofix when the indentation can be corrected safely.

### What type of rule is this?

Enforces a consistent style or user preference

### Example code

```markdown
<!-- Incorrect -->

* Item 1
  * Nested item 1
  * Nested item 2
   * Misaligned item

<!-- Correct -->

* Item 1
  * Nested item 1
  * Nested item 2
  * Nested item 3
```

### Prior Art

- markdownlint: [`MD005` - Inconsistent indentation for list items at the same level](https://github.com/DavidAnson/markdownlint/blob/main/doc/md005.md)
